tests/waku_relay/test_protocol.nim

```diff
@@ -0,0 +1,1288 @@
+      
+      # When sending the 1023KiB message
+      handlerFuture = newPushHandlerFuture()
+      otherHandlerFuture = newPushHandlerFuture()
+      discard await node.publish(pubsubTopic, msg4)
+
+      # Then the message is received in both nodes
+      check:
+        await handlerFuture.withTimeout(3.seconds)
+        await otherHandlerFuture.withTimeout(3.seconds)
+        (pubsubTopic, msg4) == handlerFuture.read()
+        (pubsubTopic, msg4) == otherHandlerFuture.read()
+
+      # When sending the 1024KiB message
+      handlerFuture = newPushHandlerFuture()
+      otherHandlerFuture = newPushHandlerFuture()
+      discard await node.publish(pubsubTopic, msg5)
+
+      # Then the message is received in both nodes
+      check:
+        await handlerFuture.withTimeout(3.seconds)
+        not await otherHandlerFuture.withTimeout(3.seconds)
+        (pubsubTopic, msg5) == handlerFuture.read()
+
+      # When sending the 1025KiB message
+      handlerFuture = newPushHandlerFuture()
+      otherHandlerFuture = newPushHandlerFuture()
+      discard await node.publish(pubsubTopic, msg6)
+
+      # Then the message is received in both nodes
+      check:
+        await handlerFuture.withTimeout(3.seconds)
+        not await otherHandlerFuture.withTimeout(3.seconds)
+        (pubsubTopic, msg6) == handlerFuture.read()
+
+      # Finally stop the other node
+      await allFutures(otherSwitch.stop(), otherNode.stop())
+
+    xasyncTest "Multiple messages at once":
+      # Given a second node connected to the first one
+      let
+        otherSwitch = newTestSwitch()
+        otherNode = await newTestWakuRelay(otherSwitch)
+
+      await allFutures(otherSwitch.start(), otherNode.start())
+      let otherRemotePeerInfo = otherSwitch.peerInfo.toRemotePeerInfo()
+      check await peerManager.connectRelay(otherRemotePeerInfo)
+
+      # Given both are subscribed to the same pubsub topic
+      var otherHandlerFuture = newPushHandlerFuture()
+      var otherMessageSeq: seq[(PubsubTopic, WakuMessage)] = @[]
+      proc otherSimpleFutureHandler(topic: PubsubTopic, message: WakuMessage) {.async, gcsafe.} =
+        otherMessageSeq.add((topic, message))
+        otherHandlerFuture.complete((topic, message))
+
+      discard node.subscribe(pubsubTopic, simpleFutureHandler)
+      discard otherNode.subscribe(pubsubTopic, otherSimpleFutureHandler)
+      check:
+        node.subscribedTopics == pubsubTopicSeq
+        otherNode.subscribedTopics == pubsubTopicSeq
+      await sleepAsync(500.millis)
+
+      # When sending multiple messages from node
+      let
+        msg1 = fakeWakuMessage("msg1", pubsubTopic)
+        msg2 = fakeWakuMessage("msg2", pubsubTopic)
+        msg3 = fakeWakuMessage("msg3", pubsubTopic)
+        msg4 = fakeWakuMessage("msg4", pubsubTopic)
+      
+      discard await node.publish(pubsubTopic, msg1)
+      # FIXME: Test SEGFAULTS when publishing after first publish in pubsub.nim:handleData:317
```

Comment:
Not the same reason, but the error's origin lies in the same place as: https://github.com/waku-org/nwaku/issues/2114